File: core/src/main/java/com/apollo/core/serialize/Serializable.java
package com.apollo.core.serialize;

/**
 * write with Apollo
 * author:lsxiao
 * date:2017-05-17 17:23
 * github:https://github.com/lsxiao
 * zhihu:https://zhihu.com/people/lsxiao
 */

public interface Serializable {
    byte[] serialize(Object obj);

    <T> T deserialize(byte[] data, Class<T> clazz);
}

File: core/src/main/java/com/lsxiao/apollo/core/serialize/Serializable.kt
package com.lsxiao.apollo.core.serialize

/**
 * write with Apollo
 * author:lsxiao
 * date:2017-05-17 17:23
 * github:https://github.com/lsxiao
 * zhihu:https://zhihu.com/people/lsxiao
 */

interface Serializable {
    fun serialize(obj: Any): ByteArray

    fun <T> deserialize(data: ByteArray, clazz: Class<T>): T
}
